duckdb_engine/__init__.py

```diff
@@ -259,6 +259,32 @@ def get_view_names(
 
         return [row[0] for row in rs]
 
+    def get_schema_names(self, connection, **kw) -> Any:
+        """
+        Return unquoted database_name.schema_name unless either contains spaces or double quotes.
+        In that case, escape double quotes and then wrap in double quotes.
+        SQLAlchemy definition of a schema includes database name for databases like SQL Server (Ex: databasename.dbo)
+        (see https://docs.sqlalchemy.org/en/20/dialects/mssql.html#multipart-schema-names)
+        """
+        s = """
+            SELECT
+                CASE
+                    WHEN contains(database_name, ' ') OR contains(database_name, '"') THEN '"' || replace(database_name, '"', '""') || '"'
```

Comment:
I feel like this logic would be better expressed in the Python code? SQLAlchemy probably has a function for the escaping that would work too

Reply:
Thank you for making the change! Yes, that is much cleaner.